### ISSUE TITLE
unused-private-member bug

### ISSUE THREAD

**Issue by socketpair** [PyCQA/pylint#7204]
### Bug description

```python
class SomeClass:
    def method(self):
        self.__clsmethod()

    @staticmethod
    def __staticmethod():
        pass

    @classmethod
    def __clsmethod(cls):
        cls.__staticmethod()
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
W0238: Unused private member `SomeClass.__staticmethod()` (unused-private-member)
```

### Expected behavior

no error

### Pylint version

```shell
pylint 2.9.6
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue.

I'm going to close as a duplicate of https://github.com/PyCQA/pylint/issues/4849. The issue has been fixed in latest pylint version (probably 2.11.0).  2.9.6 is very old, you should probably upgrade :)
